checkita-core/src/main/scala/org/checkita/dqf/context/DQContext.scala

```diff
@@ -276,6 +276,7 @@ class DQContext(settings: AppSettings, spark: SparkSession, fs: FileSystem) exte
     reduceToMap(sources.map{ srcConf =>
       log.info(s"$sourceStage Reading source '${srcConf.id.value}'...")
       val source = if (readAsStream) srcConf.readStream else srcConf.read
+      if (srcConf.persist.nonEmpty) log.info(s"$sourceStage Persisting source '${srcConf.id.value}' to ${srcConf.persist.get.toString}.")
```

Comment:
We should only persist batch sources. Stream sources must not be persisted. Also, please log this message prior source reading.

In addition we need to refactor `readVirtualSources` as we now persisting sources in the source readers. Thus, we only need to log if source will be persisted there.